de.dashup.model/src/main/java/de/dashup/model/db/Database.java

```diff
@@ -201,23 +214,38 @@ public JSONArray get(Table table, Map<String, Object> whereParameters) throws SQ
         return this.get(table, whereParameters, null);
     }
 
-    public JSONArray get(Table table, Table joinOn, Map<String, Object> onParameters, Map<String, Object> whereParameters) throws SQLException {
+    /**
+     * Fetches data from database by joining two tables.
+     *
+     * @param table           right table for the join
+     * @param joinOn          left table for the join
+     * @param onParameters    columns that are used for the join. It is important that the column,which is passed as key is
+     *                        in the right table and the column that is passed as value is in the left table of the join
+     * @param whereParameters parameters that are used in the where clause to select data
+     * @return result of the database query
+     */
+    public JSONArray get(Table table, Table joinOn, Map<String, String> onParameters, Map<String, Object> whereParameters, String orderByClause) throws SQLException {
         PreparedStatement statement;
-        String query = "SELECT * FROM " + table.getTableName() + " INNER JOIN " + joinOn.getTableName() +
-                " ON ";
-        for (Map.Entry<String, Object> entry : onParameters.entrySet()) {
-            query += table.getTableName() + "." + entry.getKey() + " = " + joinOn.getTableName() + "." +entry.getValue();
+        StringBuilder query = new StringBuilder("SELECT * FROM " + table.getTableName() + " INNER JOIN " + joinOn.getTableName() +
```

Comment:
This is a potential SQL injection risk. Please use the prepared statement with the "?" as placeholders instead

Reply:
I used common way to access the db... like it is in every other method in this class...
The prepared statement will take care of the where parameters (have a look at line 237). The table names and the on parameters CAN NOT be adapted by the user cause they are "hardcoded" in the dashup service which is in the backend and completely encapsulated and isolated from the User in front of the webpage. You can ask our special technological idontknowwhat @raphaelmue about this and he will confirm this!!!

Reply:
I see this is not your fault. I will need to talk to @raphaelmue as using this.getClause(whereParameters, "WHERE", " AND ") with a whereParameters map filled with user input can lead to an SQL injection. getClause, unfortunately, uses a StringBuilder inside,which is why user input will not be treated securly by the preparedStatement. Normally you would use preparedStatement.setString(1, <your string>) to treat user input securly.